src/modules/dashboard/components/CreateEditDomainDialog/CreateEditDomainDialogForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import { ColonyRole } from '@colony/colony-js';
+import { FormikProps } from 'formik';
+import { FormattedMessage, defineMessages } from 'react-intl';
+
+import Button from '~core/Button';
+import ColorSelect from '~core/ColorSelect';
+import { Color } from '~core/ColorTag';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Input, Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import PermissionsLabel from '~core/PermissionsLabel';
+import PermissionRequiredInfo from '~core/PermissionRequiredInfo';
+
+import { Colony, useLoggedInUser } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+
+import { getAllUserRoles } from '../../../transformers';
+import { canArchitect } from '../../../users/checks';
+
+import { FormValues } from './CreateEditDomainDialog';
+import styles from './CreateEditDomainDialogForm.css';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleCreate',
+    defaultMessage: 'Create a new domain',
+  },
+  titleEdit: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleEdit',
+    defaultMessage: 'Edit domain details',
+  },
+  name: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'Domain name',
+  },
+  purpose: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'What is the purpose of this domain?',
+  },
+  annotation: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.annotation',
+    defaultMessage: 'Explain why you’re creating this domain',
+  },
+  noPermission: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.noPermission',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You need the {roleRequired} permission in {domain} to take this action.',
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+  id?: string;
+}
+
+const CreateEditDomainDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  id,
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canCreateEditDomain =
+    hasRegisteredProfile && canArchitect(allUserRoles);
+
+  return (
+    <>
+      <DialogSection>
```

Comment:
Please add the `heading` theme to just this dialog section. This way it will have the proper top margin, similar to the other modals.

```suggestion
      <DialogSection appearance={{ theme: 'heading' }}>
```

Your modal:
![Screenshot from 2021-01-06 17-11-15](https://user-images.githubusercontent.com/1193222/103784622-d9f97500-5042-11eb-9cae-01dda510b202.png)

Payment modal:
![Screenshot from 2021-01-06 17-11-36](https://user-images.githubusercontent.com/1193222/103784640-debe2900-5042-11eb-9e06-589b49123ea3.png)